artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/PagingManager.java

```diff
@@ -85,6 +86,11 @@ public interface PagingManager extends ActiveMQComponent, HierarchicalRepository
 
    void injectMonitor(FileStoreMonitor monitor) throws Exception;
 
+   /** Execute a runnable inside the PagingManager's executor */
```

Comment:
Having the default method that violates the intended contract of the method as documented seems less than ideal.  If the intention is to prevent having to add this to other PagingManager implementations that won't ever have this method invoked then throwing an UnsupportedOperationException would be better because at least then you might notice that the method is being used and not doing something on a thread that was unintended.